packages/router/useRouteQuery/index.ts

```diff
@@ -1,17 +1,17 @@
 import type { Ref } from 'vue-demi'
-import { computed, nextTick } from 'vue-demi'
-import { toValue } from '@vueuse/shared'
+import { customRef, nextTick } from 'vue-demi'
+import { toValue, tryOnScopeDispose } from '@vueuse/shared'
 import { useRoute, useRouter } from 'vue-router'
-import type { ReactiveRouteOptionsWithTransform, RouterQueryValue } from '../_types'
+import type { ReactiveRouteOptionsWithTransform, RouteQueryValueRaw } from '../_types'
 
-let _queue: Record<string, any> = {}
+let _query: Record<string, any> = {}
```

Comment:
I am not sure about having an internal global sate. Imaging the cases you have multiple Vue apps and Vue routers, they might get conflicted. Ideally if we want to do that way we could build a WeakMap binding to the route instance.